### PR TITLE
Single Sign On Auth Interface

### PR DESCRIPTION
**Pull request by nsams**
This PR extends interface for SSO plus implements required changes to use the auth methods.

Actual implementation of the interface will be in separate repositories.

### CODE REVIEW

File: Kwf/Controller/Action/User/BackendActivateController.php [hunk 1]
```diff
@@ -40,28 +48,128 @@ protected function _initFields()
...
+
+        $ns = new Kwf_Session_Namespace('kwf-backend-activate');
+        if (!$ns->state || $state != $ns->state) throw new Kwf_Exception_AccessDenied();
+
+        $state = explode('-', $state);
+        if (count($state) != 3) throw new Kwf_Exception_NotFound();
+        $authMethod = $state[0];
+        $code = $state[2];
+
+        $users = Zend_Registry::get('userModel');
```
Comment (kaufmo) [inline Kwf/Controller/Action/User/BackendActivateController.php]: Use Kwf_Registry or Zend_Registry, not both, this confuses

**Comment by kaufmo** [comment]
:+1:

File: Kwf/Controller/Action/User/BackendActivateController.php [hunk 2]
```diff
@@ -30,7 +36,9 @@ public function preDispatch()
                 }
             }
         }
-        parent::preDispatch();
+        $this->view->dep = Kwf_Assets_Package_Default::getInstance('Admin');
+
+        //parent::preDispatch();
```
Comment (fraxachun) [inline Kwf/Controller/Action/User/BackendActivateController.php]: commenting out is wrong i guess

File: Kwf/User/Auth/ActivationFields.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
+<?php
+class Kwf_User_Auth_ActivationFields extends Kwf_User_Auth_Abstract implements Kwf_User_Auth_Interface_Activation
```
Comment (fraxachun) [inline Kwf/User/Auth/ActivationFields.php]: Please comment what this class does in terms of "token times out", "token is only valid one time or more often"..

**Comment by fraxachun** [comment]
:+1:

**Pull request merged**